File: python/zigzag_conversion_6.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
      if numRows == 1:
        return s
      out = [''] * numRows
      # need to zigzag, duh
      count = 0
      inc = 1
      for i, c in enumerate(s):
       out[count] += c
       if count == numRows-1:
        inc = -1
       elif count == 0:
        inc = 1
       count += inc
      ret = ""
      return ret.join(out)

File: rust/src/zigzag_conversion_6.rs
use std::usize;

use crate::Solution;
impl Solution {
    pub fn convert(s: String, num_rows: i32) -> String {
        if num_rows == 1 {
            return s;
        }
        let mut cur: i32 = 0;
        let mut inc = 1;

        let mut res = vec![String::new(); num_rows as usize];
        for c in s.chars() {
            res[cur as usize].push(c);
            if cur == 0 {
                inc = 1;
            } else if cur == num_rows - 1 {
                inc = -1;
            }
            cur += inc;
        }

        let mut out = String::new();
        for r in res {
            out.push_str(&r);
        }
        out
    }
}

